test/extensions/quic_listeners/quiche/active_quic_listener_test.cc

```diff
@@ -43,39 +47,135 @@ class ActiveQuicListenerPeer {
-    ON_CALL(filter_chain_manager_, findFilterChain(_)).WillByDefault(Return(&filter_chain_));
-    ON_CALL(filter_chain_, networkFilterFactories()).WillByDefault(ReturnRef(filter_factory_));
-    ON_CALL(listener_config_.filter_chain_factory_, createNetworkFilterChain(_, _))
-        .WillByDefault(Invoke([](Network::Connection& connection,
-                                 const std::vector<Network::FilterFactoryCb>& filter_factories) {
+
+    quic_listener_ = std::make_unique<ActiveQuicListener>(
+        *dispatcher_, connection_handler_, listen_socket_, listener_config_, quic_config_);
+    simulated_time_system_.sleep(std::chrono::milliseconds(100));
+  }
+
+  void ConfigureMocks(int connection_count) {
+    EXPECT_CALL(listener_config_, filterChainManager())
+        .Times(connection_count)
+        .WillRepeatedly(ReturnRef(filter_chain_manager_));
+    EXPECT_CALL(listener_config_, filterChainFactory()).Times(connection_count);
+    EXPECT_CALL(listener_config_.filter_chain_factory_, createNetworkFilterChain(_, _))
+        .Times(connection_count)
+        .WillRepeatedly(Invoke([](Network::Connection& connection,
+                                  const std::vector<Network::FilterFactoryCb>& filter_factories) {
           EXPECT_EQ(1u, filter_factories.size());
           Server::Configuration::FilterChainUtility::buildFilterChain(connection, filter_factories);
           return true;
         }));
+    EXPECT_CALL(network_connection_callbacks_, onEvent(Network::ConnectionEvent::Connected))
+        .Times(connection_count);
+    EXPECT_CALL(network_connection_callbacks_, onEvent(Network::ConnectionEvent::LocalClose))
+        .Times(connection_count);
 
-    quic_listener_ = std::make_unique<ActiveQuicListener>(
-        *dispatcher_, connection_handler_, listen_socket_, listener_config_, quic_config_);
-    simulated_time_system_.sleep(std::chrono::milliseconds(100));
+    testing::Sequence seq;
+    for (int i = 0; i < connection_count; ++i) {
+      auto read_filter = std::make_shared<Network::MockReadFilter>();
+      filter_factories_.push_back(
+          {Network::FilterFactoryCb([read_filter, this](Network::FilterManager& filter_manager) {
+            filter_manager.addReadFilter(read_filter);
+            read_filter->callbacks_->connection().addConnectionCallbacks(
+                network_connection_callbacks_);
+          })});
+      // Stop iteration to avoid calling getRead/WriteBuffer().
+      EXPECT_CALL(*read_filter, onNewConnection())
+          .WillOnce(Return(Network::FilterStatus::StopIteration));
+      read_filters_.push_back(std::move(read_filter));
+
+      filter_chains_.emplace_back();
+      EXPECT_CALL(filter_chains_.back(), networkFilterFactories())
+          .WillOnce(ReturnRef(filter_factories_.back()));
+
+      // A Sequence must be used to allow multiple EXPECT_CALL().WillOnce()
+      // calls for the same object.
+      EXPECT_CALL(filter_chain_manager_, findFilterChain(_))
```

Comment:
How about moving this EXPECT_CALL outside of for loop and use Times(connection_count)?

Reply:
Unfortunately that would not work, because findFilterChain() needs to return a different filter chain each time.